deploy/cdkpipeline/cdkpipeline.py

```diff
@@ -0,0 +1,147 @@
+        codestar_arn = ssm.StringParameter.from_string_parameter_attributes(self, "codestarArn",
+            parameter_name="/sscheck/codestar-connection-arn"
+        ).string_value
+
+        cloud_artifact = codepipeline.Artifact(
+            artifact_name="sscheck_pipeline_cloud"
+        )
+
+        source_artifact = codepipeline.Artifact(
+            artifact_name="sscheck_pipeline_source",
+        )
+
+        # Fetch github repository for changes
+        code_star_action = codepipeline_actions.CodeStarConnectionsSourceAction(
+            connection_arn = codestar_arn,
+            output = source_artifact,
+            owner = "umccr",
+            repo = "samplesheet-check-frontend",
+            branch = "implement-pipeline",
+            action_name = "Source"
+        )
+
+
+        # Create CDK pipeline
+        pipeline = pipelines.CdkPipeline(
+            self,
+            "CDKPipeline",
+            cloud_assembly_artifact = cloud_artifact,
+            pipeline_name="CDKSampleSheetCheckFrontEnd",
+            source_action = code_star_action,
+            synth_action = pipelines.SimpleSynthAction(
+                synth_command = "cdk synth",
+                cloud_assembly_artifact = cloud_artifact,
+                source_artifact = source_artifact,
+                install_commands = [
+                    "npm install -g aws-cdk",
+                    "gem install cfn-nag",
+                    "pip install -r requirements.txt"
+                ],
+                test_commands = [
+                    "mkdir ./cfnnag_output",
+                    "for template in $(find ./cdk.out -type f -maxdepth 2 -name '*.template.json'); do cp $template ./cfnnag_output; done",
+                    "cfn_nag_scan --input-path ./cfnnag_output"
+                ],
+                action_name = "Synth",
+                project_name = "cdk_synth",
+                subdirectory = "deploy"
+            )
+
+        )
+            
+
+        # Deploy infrastructure
+        pipeline.add_application_stage(
+            SampleSheetCheckStage(
+                self,
+                "SampleSheetCheckFrontEndStage",
+            )
+        )
+
+        react_build_stage = pipeline.add_stage(
+            stage_name="ReactBuild",
+        )
+        front_end_bucket_name=ssm.StringParameter.from_string_parameter_attributes(
+            self,
+            "FrontEndBucketName",
+            parameter_name="/sscheck/bucket_name"
+        ).string_value
+
+        front_end_bucket_arn = s3.Bucket.from_bucket_name(
+            self,
+            "FrontEndBucket",
+            bucket_name=front_end_bucket_name
+        ).bucket_arn
+
+        react_build_stage.add_actions(
+            pipelines.ShellScriptAction(
+                action_name = "BuildScript",
+                commands = [
+                    "export REACT_APP_BUCKET_NAME=$(aws ssm get-parameter --name '/sscheck/bucket_name' | jq -r .Parameter.Value)",
```

Comment:
`ShellScriptAction` can take an `environment_variables` parameter. Would that not be easier/clearer here?

Reply:
Yes, thanks for pointing it out. Changed it on commit `a6afdcc`

Did not see the option in detail. It could fetch directly to the parameter and could mask data when printed at the build log.